lib/solveEquation/stepThrough.js

```diff
@@ -58,8 +58,11 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
     steps = addSimplificationSteps(steps, equation, debug);
     if (steps.length > 0) {
       const lastStep = steps[steps.length - 1];
-      equation = Equation.createEquationFromString(
-        lastStep.newEquation.print(), equation.comparator);
+      equation.leftNode = lastStep.newEquation.leftNode.clone();
+      equation.rightNode = lastStep.newEquation.rightNode.clone();
+      // console.log(left);
+      // equation = Equation.createEquationFromString(
+      //   lastStep.newEquation.print(), equation.comparator);
```

Comment:
This is kind of a big change.  I'm not sure how much anyone relies on being able to re-parse the output of `print` beyond what was being done in `stepThrough`.  I could see having some kind of serialization/deserialization method being useful though.